operator/internal/webhook/admission/pcs/authorization/handler.go

```diff
@@ -18,18 +18,144 @@ package authorization
+	groveconfigv1alpha1 "github.com/NVIDIA/grove/operator/api/config/v1alpha1"
+	k8sutils "github.com/NVIDIA/grove/operator/internal/utils/kubernetes"
 
 	"github.com/go-logr/logr"
+	admissionv1 "k8s.io/api/admission/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
 // Handler is the PodCliqueSet authorization admission webhook handler.
 type Handler struct {
-	Logger  logr.Logger
-	Decoder admission.Decoder
+	config  groveconfigv1alpha1.AuthorizerConfig
+	client  client.Client
+	decoder *requestDecoder
+	logger  logr.Logger
+}
+
+// NewHandler creates a new handler for the auhtorizer webhook.
+func NewHandler(mgr manager.Manager, config groveconfigv1alpha1.AuthorizerConfig) *Handler {
+	return &Handler{
+		config:  config,
+		client:  mgr.GetClient(),
+		decoder: newRequestDecoder(mgr),
+		logger:  mgr.GetLogger().WithName("authorizer-webhook").WithName(Name),
+	}
 }
 
 // Handle handles requests and admits them if they are authorized.
-func (h *Handler) Handle(_ context.Context, _ admission.Request) admission.Response {
-	return admission.Response{}
+func (h *Handler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	log := h.logger.WithValues("user", req.UserInfo.Username, "operation", req.Operation, "resource", req.Resource, "subresource", req.SubResource, "name", req.Name, "namespace", req.Namespace)
+	log.Info("Authorizer webhook invoked")
+
+	// always allow `CONNECT` operations, irrespective of the user.
+	if req.Operation == admissionv1.Connect {
+		log.Info("Connect operation requested, which is always allowed. Admitting.")
+		return admission.Allowed(fmt.Sprintf("operation %s is allowed", req.Operation))
+	}
+
+	// Decode and convert the request object to `metav1.PartialObjectMeta`.
+	resourcePartialObjMeta, err := h.decoder.decode(log, req)
+	if err != nil {
+		return admission.Errored(toAdmissionError(err), err)
+	}
+	resourceObjectKey := client.ObjectKeyFromObject(resourcePartialObjMeta)
+
+	// Get the parent PodCliqueSet
+	pcsPartialObjectMetadata, warnings, err := h.getParentPCSPartialObjectMetadata(ctx, resourcePartialObjMeta.ObjectMeta)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err).WithWarnings(warnings...)
+	}
+	// PCS can be nil if the resource has no reference of a PodCliqueSet or the referenced PodCliqueSet is not found.
+	if pcsPartialObjectMetadata == nil {
+		return admission.Allowed(fmt.Sprintf("admission allowed, no PodCliqueSet could be determined for resource: %v", resourceObjectKey)).WithWarnings(warnings...)
+	}
+
+	// Check if protection of PodCliqueSet managed resources has been disabled.
+	if metav1.HasAnnotation(pcsPartialObjectMetadata.ObjectMeta, apiconstants.AnnotationDisableManagedResourceProtection) {
```

Comment:
what if the value of the annotation is set to true/false? I know that you want to use a marker annotation but you also do not stop from anyone from giving a value for the annotation key.